lib/net.js

```diff
@@ -827,7 +827,8 @@ function connect(self, address, port, addressType, localAddress, localPort) {
     var sockname = self._getsockname();
     var details;
 
-    if (sockname) {
+    if (sockname && sockname.address != undefined
```

Comment:
It doesn't look like `_getsockname()` would set them to `null`. It might be useful for debugging purposes to see if they were ever explicitly set to `null`.